kernel/init.c

```diff
@@ -138,3 +138,25 @@ int create_stdio(const char *file) {
     current->files->files[2] = fd_retain(fd);
     return 0;
 }
+
+static struct fd *open_fd_from_actual_fd(int fd_no) {
+    if (fd_no < 0)
+        return ERR_PTR(errno_map());
+    struct fd *fd = fd_create(&realfs_fdops);
```

Comment:
Should use adhoc_fd_create, otherwise filesystem operations like fstat will crash.

Also this could return an error which should be propagated.